receiver/stanzareceiver/receiver_test.go

```diff
@@ -72,3 +80,107 @@ func TestHandleConsumeError(t *testing.T) {
 	receiver.Shutdown(context.Background())
 	require.Equal(t, 1, mockConsumer.rejected, "one log entry expected")
 }
+
+func BenchmarkPipelineSimple(b *testing.B) {
+
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		b.Errorf(err.Error())
+		b.FailNow()
+	}
+
+	filePath := filepath.Join(tempDir, "bench.log")
+
+	pipelineYaml := fmt.Sprintf(`
+- type: file_input
+  include:
+    - %s
+  start_at: beginning`,
+		filePath)
+
+	pipelineCfg := pipeline.Config{}
+	require.NoError(b, yaml.Unmarshal([]byte(pipelineYaml), &pipelineCfg))
+
+	emitter := NewLogEmitter(zap.NewNop().Sugar())
+	defer emitter.Stop()
+
+	buildContext := testutil.NewBuildContext(b)
+	buildContext.Logger = zap.NewNop().Sugar() // be quiet
+
+	pl, err := pipelineCfg.BuildPipeline(buildContext, nil, emitter)
+	require.NoError(b, err)
+
+	// Populate the file that will be consumed
+	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE, 0666)
+	require.NoError(b, err)
+	for i := 0; i < b.N; i++ {
+		file.WriteString("testlog\n")
```

Comment:
I am not sure I fully understand what we benchmark. Looks like we are writing N lines to a file and then read them all. Does this benchmark measure the time it takes to write and read a single line on average?

Reply:
It should be measuring how long it takes to read and transform a single line on average, but I was missing the call to `convert`, which I'll add momentarily.

As you said, we're writing N lines to a file, but then we are:
- resetting the benchmark timer to exclude time spent writing
- starting the stanza pipeline, which begins consuming the file
- as logs are tailed, convert them from stanza format to `pdata.Logs`

```
// Run the actual benchmark
b.ResetTimer()
require.NoError(b, pl.Start())
for i := 0; i < b.N; i++ {
	convert(<-emitter.logChan)
}
```

To be clear, I think Ben pointed out that there is room for improvement in terms of efficiency here. I think the current implementation is acceptable though and wanted to optimize for performance in a future PR.